File: app/src/main/java/com/gdi/hahi/mvp/model/LiveModel.java
package com.gdi.hahi.mvp.model;

import android.content.Context;

import com.gdi.hahi.mvp.contract.HahiContract;
import com.gdi.hahi.mvp.model.bean.LiveBean;
import com.gdi.hahi.mvp.model.bean.Neihan;
import com.gdi.hahi.mvp.model.bean.SimpleLiveBean;
import com.gdi.hahi.network.Api;
import com.gdi.hahi.network.RetrofitClient;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;

/**
 * Created by gandi on 2017/9/8 0008.
 */

public class LiveModel implements HahiContract.Model<LiveBean,List<SimpleLiveBean>> {

    String device_id;

    public LiveModel(String device_id) {
        this.device_id = device_id;
    }

    @Override
    public List<SimpleLiveBean> convertToDoMain(LiveBean data) {
        List<SimpleLiveBean> list = new ArrayList<>();

        List<LiveBean.DataBeanX> datal =  data.getData();
        for (LiveBean.DataBeanX beans :datal){
            SimpleLiveBean lBean = new SimpleLiveBean();
            LiveBean.DataBeanX.DataBean bean = beans.getData();
            lBean.setUser_name(bean.getOwner().getNickname());
            lBean.setAvatar_url(bean.getOwner().getAvatar_medium().getUrl_list().get(0));
            lBean.setAvatar_jpg(bean.getOwner().getAvatar_jpg().getUrl_list().get(0));
            lBean.setCity(bean.getOwner().getCity());
            lBean.setRtmp_pull_url(bean.getStream_url().getRtmp_pull_url());
            list.add(lBean);
        }

        return list;
    }

    @Override
    public Observable<LiveBean> loadData(Context context, long lastTime) {
        RetrofitClient retrofitClient = RetrofitClient.getInstance(context, Api.LiveBaseUrl);
        Observable<LiveBean> observable = retrofitClient.create(Api.class).getLiveData(device_id,170,lastTime);
        return observable;
    }
}

File: app/src/main/java/com/gdi/hahi/mvp/model/LiveModel.kt
package com.gdi.hahi.mvp.model

import android.content.Context

import com.gdi.hahi.mvp.contract.HahiContract
import com.gdi.hahi.mvp.model.bean.DataL
import com.gdi.hahi.mvp.model.bean.LData
import com.gdi.hahi.mvp.model.bean.LiveBean
import com.gdi.hahi.mvp.model.bean.LiveData
import com.gdi.hahi.network.Api
import com.gdi.hahi.network.RetrofitClient

import java.util.ArrayList

import io.reactivex.Observable

/**
 * Created by gandi on 2017/9/8 0008.
 */

class LiveModel(internal var device_id: String) : HahiContract.Model<LiveBean, List<LiveData>> {

    override fun convertToDoMain(data: LiveBean): List<LiveData>? {
        if (data == null)
            return null

        return data.data.map { convertLiveDataToDoMain(it) }
    }


    fun convertLiveDataToDoMain(data: LData): LiveData {
        with(data.data) {
            return LiveData(owner.nickname, owner.avatar_medium.url_list[0], owner.avatar_jpg.url_list[0], owner.city, stream_url.rtmp_pull_url)
        }
    }

    override fun loadData(context: Context, lastTime: Long): Observable<LiveBean> {
        val retrofitClient = RetrofitClient.getInstance(context, Api.LiveBaseUrl)
        val observable = retrofitClient.create<Api>(Api::class.java).getLiveData(device_id, 170, lastTime)
        return observable
    }
}
